cmd/spire-server/cli/federation/list_test.go

```diff
@@ -0,0 +1,138 @@
+
+	require.Equal(t, `Usage of federation list:
+  -socketPath string
+    	Path to the SPIRE Server API socket (default "/tmp/spire-server/private/api.sock")
+`, test.stderr.String())
+}
+
+func TestShowSynopsis(t *testing.T) {
+	test := setupTest(t, newListCommand)
+	require.Equal(t, "List federation relationships", test.client.Synopsis())
+}
+
+func TestList(t *testing.T) {
+	federation1 := &types.FederationRelationship{
+		TrustDomain:           "foh.test",
+		BundleEndpointUrl:     "https://foo.test/endpoint",
+		BundleEndpointProfile: &types.FederationRelationship_HttpsWeb{},
+	}
+
+	federation2 := &types.FederationRelationship{
+		TrustDomain:       "bar.test",
+		BundleEndpointUrl: "https://bar.test/endpoint",
+		BundleEndpointProfile: &types.FederationRelationship_HttpsSpiffe{
+			HttpsSpiffe: &types.HTTPSSPIFFEProfile{
+				EndpointSpiffeId: "spiffe://bar.test/id",
+				Bundle:           &types.Bundle{TrustDomain: "bar.test"},
+			},
+		},
+	}
+	federation3 := &types.FederationRelationship{
+		TrustDomain:       "baz.test",
+		BundleEndpointUrl: "https://baz.test/endpoint",
+		BundleEndpointProfile: &types.FederationRelationship_HttpsSpiffe{
+			HttpsSpiffe: &types.HTTPSSPIFFEProfile{
+				EndpointSpiffeId: "spiffe://baz.test/id",
+			},
+		},
+	}
+
+	for _, tt := range []struct {
+		name string
+		arg  []string
+
+		expectListReq *trustdomainv1.ListFederationRelationshipsRequest
+		listResp      *trustdomainv1.ListFederationRelationshipsResponse
+
+		serverErr error
+
+		expectOut string
+		expectErr string
+	}{
+		{
+			name:          "no federations",
+			expectListReq: &trustdomainv1.ListFederationRelationshipsRequest{},
+			listResp:      &trustdomainv1.ListFederationRelationshipsResponse{},
+			expectOut:     "Found 0 federation relationships\n",
+		},
+		{
+			name:          "single federation",
+			expectListReq: &trustdomainv1.ListFederationRelationshipsRequest{},
+			listResp: &trustdomainv1.ListFederationRelationshipsResponse{
+				FederationRelationships: []*types.FederationRelationship{federation1},
+			},
+			expectOut: `Found 1 federation relationship
+
+Trust domain              : foh.test
+Bundle endpoint URL       : https://foo.test/endpoint
+Bundle endpoint profile   : https_web
+`,
+		},
+		{
+			name:          "multiple federations",
+			expectListReq: &trustdomainv1.ListFederationRelationshipsRequest{},
+			listResp: &trustdomainv1.ListFederationRelationshipsResponse{
+				FederationRelationships: []*types.FederationRelationship{
+					federation1,
+					federation2,
+					federation3,
+				},
+			},
+			expectOut: `Found 3 federation relationships
+
+Trust domain              : foh.test
+Bundle endpoint URL       : https://foo.test/endpoint
+Bundle endpoint profile   : https_web
+
+Trust domain              : bar.test
+Bundle endpoint URL       : https://bar.test/endpoint
+Bundle endpoint profile   : https_spiffe
+Endpoint SPIFFE ID        : spiffe://bar.test/id
+Bundle trust domain       : bar.test
```

Comment:
I don't think we should show this. The bundle provided in the https_spiffe profile should always be the bundle of the trust domain for which the relationship is for. I don't recall if our handlers are validating this correctly. I will look. But either way, this field is them superfluous.

Relevant spec text:
https://github.com/spiffe/spiffe/blob/main/standards/SPIFFE_Federation.md#5222-endpoint-parameters

See the text about self-serving v.s. non self-serving (and related examples).